packages/backend/src/auth/middleware.ts

```diff
@@ -14,14 +18,30 @@ export function authenticate(
   const { token } = req.cookies
   if (token) {
     const secretKey = config.auth.jwtSecretKey
-    const verification = jwt.verify(token, secretKey)
-    if (typeof verification === 'string') {
-      next(
-        new Error(`Malformed JWT token: received 'string', expected 'object'`),
-      )
-    } else {
-      ;(req as RequestWithToken).jwtPayload = verification
-      next()
+
+    try {
+      const verification = jwt.verify(token, secretKey)
+
+      if (typeof verification === 'string') {
+        next(
+          new Error(
+            `Malformed JWT token: received 'string', expected 'object'`,
+          ),
+        )
+      } else {
+        ;(req as RequestWithToken).jwtPayload = verification
+        next()
+      }
+    } catch (error) {
+      if (error instanceof TokenExpiredError) {
+        res.status(403).json({ error: 'Your token has expired.' })
+      } else if (error instanceof NotBeforeError) {
+        res.status(403).json({ error: 'Current time is before the nbf claim.' })
+      } else if (error instanceof JsonWebTokenError) {
+        res.status(403).json({ error: 'Invalid signature.' })
+      } else {
+        res.status(403).json({ error: 'Unknown token error.' })
```

Comment:
Perhaps in this case, since it's unknown, we can just throw, and it can be handled by the error handler middleware? i.e. return 500 server error

Reply:
Yes. This should be 500. Also the error `You are not logged in.` below should be perhaps 401